src/locale/localed-util.c

```diff
@@ -142,6 +146,13 @@ int x11_read_data(Context *c, sd_bus_message *m) {
                 c->x11_cache = sd_bus_message_ref(m);
         }
 
+        /* Check if there is X11 keyboard data inside vconsole */
+        r = vconsole_read_data(c, m);
+        if(r <= 0 && c->x11_layout != NULL && errno != ENOENT) {
+            if(r == 0) log_debug("Found X11 keyboard data inside vconsole, skipping /etc/X11/xorg.conf.d/00-keyboard.conf");
+                return r;
+        }
```

Comment:
Noted, thanks for the pointers :+1: 

I was checking for `errno` to prevent the X11 layout from being set if `vconsole.conf` does not exist and instead read the xorg config file. I should check `r` instead

Reply:
Also, since I call `vconsole_read_data` inside `x11_read_data`, should I call `vconsole_write_data` inside `x11_write_data`?

Reply:
> Also, since I call `vconsole_read_data` inside `x11_read_data`, should I call `vconsole_write_data` inside `x11_write_data`?

I do not think so, as the error should be propagated to the DBus client.

Reply:
ok